### DISCUSSION TITLE
There's no way to run this in the background without leaving it open on the taskbar?

### DISCUSSION THREAD

**Discussion by brianjlowry** [philosowaffle/peloton-to-garmin#612] [Ideas]
Could it be possible to just create a console version so that we can leverage something like AlwaysUp to have it run in the background and sync automatically? I don't understand why the decision was made to eliminate the ability for it to run in the background...

**Comment by philosowaffle** [comment]
There are currently [5 different ways](https://philosowaffle.github.io/peloton-to-garmin/latest/install/) to run P2G.  All of them support some form of "running in the background" except for the Windows executable.  

The windows exe has never _natively_ supported running in the background, though it sounds like you maybe found a workaround using a 3rd party tool.  Honestly, I had no idea anyone was doing this.

The guiding principle for the Windows exe is to increase the accessibility of P2G for the average Peloton person who's never heard of JSON, or C#, or even GitHub for that matter.  With V4 I wanted to give these folks a GUI Application that let them configure and run P2G in a much more familiar environment.

But I can understand that the background sync feature is important to you so I'd recommend checking out the [GitHub Action option](https://philosowaffle.github.io/peloton-to-garmin/latest/install/github-action/). It should give you feature parity without the need to run an extra background service on your personal computer.

**Reply by brianjlowry** [reply]
For me, having the UI minimize to an icon in the notification area would be the perfect solution. I just don't like taskbar windows open, if I can avoid it. I will close them by accident often... I'm trying to get used to it.

AlwaysUp worked well with the old .exe (pre-UI), but I fully support the move to the UI. AlwaysUp still works with the new UI, but it displays the UI instead of just hiding the console app in the background.

If it can't be turned into a dual UI/console app, then I'm just curious if it could be hidden to the notification area so it can sync in the background, and I don't have to manually push exercises every time. It's such a clutch program.

I appreciate you!

**Reply by geoffbon** [reply]
@brianjlowry I'm not sure which version of Windows you're on, but if the GitHub Actions solution is more than you want to tackle then Windows 11 lets you have multiple desktops; you could just move the app to a different desktop? You won't see it unless you switch to that desktop.  

I'm _really_ looking forward to using one of the other methods so that I can leave it running unattended (Docker container, hosted in Kubernetes), but for me they all need the MFA issue resolved (#585).  I'm assuming this applies to anyone else with a watch that has ECG functionality, in my case the Epix Gen 2 Pro, since this forces two-step verification.

All that said, I'm going to assume there's quite a large number of Peloton users for whom the other methods of deployment are too technical, but who will get a lot of value out of a background sync to Garmin once the MFA thing is sorted.  Something like the [WeatherTwentyOne sample](https://github.com/dotnet/maui-samples/tree/main/8.0/Apps/WeatherTwentyOne/src/WeatherTwentyOne/Platforms/Windows)?

**Reply by brianjlowry** [reply]
@geoffbon - I am on Windows 10. I will take a look at the other methods. Thanks!
